adam/access_calculation.py

```diff
@@ -0,0 +1,201 @@
+"""
+    access_calculation.py
+"""
+from adam.adam_objects import AdamObject
+from adam.adam_objects import AdamObjects
+from adam.batch_propagation import SinglePropagation
+from adam.opm_params import OpmParams
+from adam.propagation_params import PropagationParams
+
+from datetime import datetime, timedelta
+
+
+class AccessCalculation(AdamObject):
+    def __init__(self, propagation_params, opm_params, asteroid_propagation_uuid,
+                 access_start_time, access_end_time, pointings_table_name):
+        """
+            asteroid_propagation_uuid (string): UUID of SinglePropagation to use as
+                asteroid location. Either asteroid_propagation_uuid or propagation_params/opm_params
+                are required.
+            access_start_time (string): Start time of interval in which accesses should be computed.
+                (ISO format, UTC). Required!
+            access_end_time (string): End time of interval in which accesses should be computed.
+                (ISO format, UTC). Required!
+            pointings_table_name (string): Name of the sql table in which the telescope
+                pointings are stored.
+        """
+        AdamObject.__init__(self)
+        self._propagation_params = propagation_params
+        self._opm_params = opm_params
+        self._asteroid_propagation_uuid = asteroid_propagation_uuid
+        self._access_start_time = access_start_time
+        self._access_end_time = access_end_time
+        self._pointings_table_name = pointings_table_name
+        self._accesses = None
+
+    # Accesses format:
+    #    List of [JD start, JD end, datetime start, datetime end]
+    def set_accesses(self, accesses):
+        self._accesses = accesses
+
+    def _parse_isoformat_date(self, date):
+        # TODO(jcarrico): Find a more precise datetime library.
+        parts = date.split(".")
+        micros = float("0." + parts[1].replace('Z', ''))
+
+        parsed = datetime.strptime(parts[0], "%Y-%m-%dT%H:%M:%S")
+        parsed = parsed + timedelta(microseconds=micros * 1000000)
+        return parsed
+
+    def set_accesses_from_str(self, accesses):
+        if accesses is None:
+            self._accesses = []
```

Comment:
Optional: `self._accesses = []` can be before if, and then if would check for not None, without else.